lib/nostrum/struct/channel.ex

```diff
@@ -37,132 +135,282 @@ defmodule Nostrum.Struct.Channel do
+  """
+  @typedoc since: "0.5"
+  @type rtc_region :: String.t() | nil
 
+  @typedoc """
+  The video quality mode of the voice channel.
+
+  More information about _video quality modes_ can be found on the [Discord API Video Quality Mode Documentation](https://discord.com/developers/docs/resources/channel#channel-object-video-quality-modes).
   """
-  @type type :: integer()
+  @typedoc since: "0.5"
+  @type video_quality_mode :: integer() | nil
+
+  @typedoc """
+  Approximate count of messages in a thread, capped at 50.
+  """
+  @typedoc since: "0.5"
+  @type message_count :: integer() | nil
+
+  @typedoc """
+  Approximate count of members in a thread, capped at 50.
+  """
+  @typedoc since: "0.5"
+  @type member_count :: integer() | nil
+
+  @typedoc """
+  The video quality mode of the voice channel.
+
+  More information about _video quality modes_ can be found on the [Discord API Video Quality Mode Documentation](https://discord.com/developers/docs/resources/channel#channel-object-video-quality-modes).
+  """
+  @typedoc since: "0.5"
+  @type thread_metadata :: %{
+          archived: archived,
+          auto_archive_duration: auto_archive_duration,
+          archive_timestamp: archive_timestamp,
+          locked: boolean()
+        }
+
+  @typedoc """
+  When the thread was archived.
+  """
+  @typedoc since: "0.5"
+  @type archive_timestamp :: String.t() | nil
+
+  @typedoc """
+  The threads locked status.
+  """
+  @typedoc since: "0.5"
+  @type locked :: boolean()
+
+  @typedoc """
+  The threads archived status.
+  """
+  @typedoc since: "0.5"
+  @type archived :: boolean()
+
+  @typedoc """
+  Archive duration for the thread.
+
+  - 60, 1 hour
+  - 1440, 24 hours
+  - 4320, 3 days
+  - 10080, 7 days
+  """
+  @typedoc since: "0.5"
+  @type auto_archive_duration :: integer()
+
+  @typedoc """
+  Indicates whether a user has joined a thread or not.
+  """
+  @typedoc since: "0.5"
+  @type member :: %{
+          id: id,
+          user_id: user_id,
+          join_timestamp: join_timestamp,
+          flags: flags
+        }
+
+  @typedoc """
+  User id of the threads creator.
+  """
+  @typedoc since: "0.5"
+  @type user_id :: Snowflake.t()
+
+  @typedoc """
+  When the current user joined the thread.
+  """
+  @typedoc since: "0.5"
+  @type join_timestamp :: String.t()
 
   @typedoc """
-  A `Nostrum.Struct.Channel` that represents a text channel in a guild
+  User thread settings, currently only used for notifications.
+  """
+  @typedoc since: "0.5"
+  @type flags :: integer()
+
+  @typedoc """
+  Default duration for newly created threads.
+
+  - 60, 1 hour
+  - 1440, 24 hours
+  - 4320, 3 days
+  - 10080, 7 days
+  """
+  @typedoc since: "0.5"
+  @type default_auto_archive_duration :: integer()
+
+  @typedoc """
+  Computed permissions of the invoking user.t()
```

Comment:
Would it make more sense to just write "user" here? There's no relation to the struct.

```suggestion
  Computed permissions of the invoking user.
```